### ISSUE TITLE
Dictation voice input

### ISSUE THREAD

**Issue by Pecka95** [microsoft/Microsoft-OpenXR-Unreal#17]
Hey! I'm trying to using this plugin to add voice input as dictation into game. I don't understand, how to send recognized phrase into my character. Can you help me?

**Comment by sw5813** [comment]
Check out the Microsoft docs for instructions: https://docs.microsoft.com/en-us/windows/mixed-reality/develop/unreal/unreal-voice-input

**Comment by Pecka95** [comment]
> Check out the Microsoft docs for instructions: https://docs.microsoft.com/en-us/windows/mixed-reality/develop/unreal/unreal-voice-input

Yeah, I saw this docs and it's not what i'm asking for. I want to get a recognized phrase from plugin not by speech mapping. I'm trying to send phrase in string form

**Comment by sw5813** [comment]
Gotcha! While dictation isn't supported by the Microsoft OpenXR plugin out of the box, dictation is supported via the SpeechRecognizer WinRT API. We use the same WinRT API to get keywords in the Microsoft OpenXR plugin, so one option would be for you to copy/paste our code and make some small changes to use dictation instead of keywords, then surface that dictated text to your game. 

Here is the Microsoft OpenXR plugin code that calls the WinRT speech recognizer with a Keywords constraint in order to recognize keywords: https://github.com/microsoft/Microsoft-OpenXR-Unreal/blob/main/MsftOpenXRGame/Plugins/MicrosoftOpenXR/Source/MicrosoftOpenXR/Private/SpeechPlugin.cpp#L203

The ResultGenerated just has the keyword: https://github.com/microsoft/Microsoft-OpenXR-Unreal/blob/main/MsftOpenXRGame/Plugins/MicrosoftOpenXR/Source/MicrosoftOpenXR/Private/SpeechPlugin.cpp#L248

Here is an example of code that uses that same SpeechRecognizer with a dictation constraint: 
https://github.com/microsoft/Windows-universal-samples/blob/master/Samples/SpeechRecognitionAndSynthesis/cpp/Scenario_ContinuousDictation.xaml.cpp#L101
 
Then the ResultGenerated callback will have the dictated text instead of the keyword: https://github.com/microsoft/Windows-universal-samples/blob/master/Samples/SpeechRecognitionAndSynthesis/cpp/Scenario_ContinuousDictation.xaml.cpp#L353

**Comment by Pecka95** [comment]
Thank you!! It looks like exactly what i'm looking for! I'll try this. And thank you for a such detailed guide!

**Comment by sw5813** [comment]
Glad to be able to help!

**Comment by FabianEP11** [comment]
Hi, I am trying to do the same as @Pecka95 described: I need a functionality as the dictation mode (not speech mapping), but I'm not sure how to include it on my app.  
I was following the files you provided and I was able to test the _speechRecognitionAndSyntesis_ app on the Hololens device and it is exactly what I need; however, I'm having hard time figuring out how to include it on an app created directly on Unreal.
Could you please guide me with this? or does it work just for UWP apps?

**Comment by frank-Regal** [comment]
@Pecka95, have you been able to successful implement the above approach? We are looking for a similar implementation
